internal/encoding/types.go

```diff
@@ -410,38 +410,38 @@ func (f *Filter) Unmarshal(r *buffer.Buffer) error {
 
 // peekMessageType reads the message type without
 // modifying any data.
-func PeekMessageType(buf []byte) (uint8, error) {
+func PeekMessageType(buf []byte) (uint8, uint8, error) {
 	if len(buf) < 3 {
-		return 0, errors.New("invalid message")
+		return 0, 0, errors.New("invalid message")
 	}
 
 	if buf[0] != 0 {
-		return 0, fmt.Errorf("invalid composite header %02x", buf[0])
+		return 0, 0, fmt.Errorf("invalid composite header %02x", buf[0])
 	}
 
 	// copied from readUlong to avoid allocations
 	t := AMQPType(buf[1])
 	if t == TypeCodeUlong0 {
-		return 0, nil
+		return 0, 2, nil
```

Comment:
This might be wrong, the old behaviour was to skip 3 bytes, but the header is 2 bytes long as far as I can tell.
I might be wrong though.

Reply:
The three bytes would be correct for the case where the encoded length of the message type is a `smallulong`.
```
00 - composite type prefix
83 - smallulong
xx - smallulong value
```

http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-types-v1.0-os.html#type-ulong

From your repro, application properties is encoded as an 8-byte ulong (`0, 128, 0, 0, 0, 0, 0, 0, 0, 116`).

It appears that Azure service bus encodes application properties size as a smallulong.  I don't see anything in the AMQP 1.0 spec that restricts the size to a smallulong so this should be fine.

Reply:
When unmarshalling a Message, ulong0 isn't valid as a type in that context, so even if the size is wrong here it won't matter.